extensions/wikia/WikiaRSS/WikiaRssExternalController.class.php

```diff
@@ -32,24 +34,54 @@ public function getRssFeeds() {
 				return;
 			}
 
-			$status = null;
-			$rss = @fetch_rss($url, $status);
+			$feedData = WikiaDataAccess::cacheWithLock(
+				wfSharedMemcKey( 'WikiaRss', 'feed', $url ),
+				self::CACHE_TIME,
+				function() use( $url ) {
+					\Wikia\Logger\WikiaLogger::instance()->info(
+						'WikiaRSS calling RSS provider',
+						[ 'providerUrl' => $url ]
+					);
+
+					$status = null;
+					$rss = @fetch_rss($url, $status);
+
+					$errorMessageKey = null;
+					$error = null;
+
+					if( !is_null($status) && $status !== 200 ) {
+						$errorMessageKey = 'wikia-rss-error-wrong-status-' . $status;
+					} else if( !is_object($rss) || !is_array($rss->items) ) {
+						$errorMessageKey = 'wikia-rss-empty';
+					} else if( $rss->ERROR ) {
+						$errorMessageKey = 'wikia-rss-error';
+						$error = $rss->ERROR;
+					}
 
-			if( !is_null($status) && $status !== 200 ) {
-				$this->response->setVal('error', wfMsg('wikia-rss-error-wrong-status-'.$status, $url));
-				return;
-			}
+					if ( !empty( $errorMessageKey ) ) {
+						\Wikia\Logger\WikiaLogger::instance()->error(
+							'Error: WikiaRSS calling RSS provider ',
+							[ 'providerUrl' => $url, 'error' => $errorMessageKey ]
+						);
+					}
 
-			if( !is_object($rss) || !is_array($rss->items) ) {
-				$this->response->setVal('error', wfMsg('wikia-rss-empty', $url));
-				return;
-			}
+					return [
+						'errorMessageKey' => $errorMessageKey,
+						'error' => $error,
+						'rss' => $rss
+					];
+				}
+			);
 
-			if( $rss->ERROR ) {
-				$this->response->setVal('error', wfMsg('wikia-rss-error', $url, $rss->ERROR));
+			if ( !empty( $feedData['errorMessageKey'] ) ) {
```

Comment:
nitpick:
I'd check if there is an error by checking if error is not null 
and not a message